test/NuGet.Protocol.Core.v3.Tests/RetryHandlerTests.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Core.v3.Data;
+using Xunit;
+
+namespace NuGet.Protocol.Core.v3.Tests
+{
+    public class RetryHandlerTests
+    {
```

Comment:
Add a test to verify there is a minimal delay between the retries, move the delay calculation to a method, and verify the minimal delay by calling the method

Move the retry count to be a public readonly and use that when testing the max number of retries